framework/src/modules/interoperability/sidechain/module.ts

```diff
@@ -195,4 +211,184 @@ export class SidechainInteroperabilityModule extends BaseInteroperabilityModule
 	public async init(_args: ModuleInitArgs) {
 		this._mainchainRegistrationCommand.addDependencies(this._validatorsMethod);
 	}
+
+	// @see https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#mainchain
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async initGenesisState(ctx: GenesisBlockExecuteContext): Promise<void> {
+		const genesisBlockAssetBytes = ctx.assets.getAsset(MODULE_NAME_INTEROPERABILITY);
+		if (!genesisBlockAssetBytes) {
+			return;
+		}
+
+		const genesisInteroperability = codec.decode<GenesisInteroperability>(
+			genesisInteroperabilitySchema,
+			genesisBlockAssetBytes,
+		);
+
+		validator.validate<GenesisInteroperability>(
+			genesisInteroperabilitySchema,
+			genesisInteroperability,
+		);
+
+		await this._verifyChainInfos(ctx, genesisInteroperability);
```

Comment:
I find this structure a bit confusing. I would add here the check for case of length == 0 and corresponding checks for `ownChainName`, nonce, etc. And then include in `verifyChainInfos` only checks for the case that length is 1. 

This way we will also be more consistent with the mainchain implementation and easier to debug. For example now, the `verifyTerminatedStateAccounts` fct is called inside `verifyChainInfos, which is a bit weird.